### PR TITLE
Adds docs for how parallelism works.

### PR DESCRIPTION
**Pull request by mavenraven**
@davidxia @mattnworb @caipre

### CODE REVIEW

File: docs/parallelism.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Paralellism
```
Comment (mattnworb) [inline docs/parallelism.md]: nit: this is misspelt

File: docs/parallelism.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+| A | B | C |
+|---|---|---|
+| W | W | R |
+
+Now that **A** is running again, the rolling update will continue with waiting for **B** to complete deploying successfully. Since **B** has been deploying asynchrously while the rolling update waited for **A** to finish, the time to wait is typically very short. This is the mechanism that makes `--parallelism` decrease the time it takes for a deployment to finish.
+
+| A | B | C |
+|---|---|---|
+| R | W | R |
```
Comment (mattnworb) [inline docs/parallelism.md]: maybe this diagram would show more data if you labeled it as "R2" or "R1" for which version is running?

File: docs/parallelism.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Paralellism
+
+`--parallelism` is a flag available on `rolling-update`.  This flag is better thought of as "chunking" a deployment, instead
+of true paralellism.
+
+Lets say we have hosts **A**, **B**, and **C**, each in the running state:
```
Comment (davidxia) [inline docs/parallelism.md]: nit: Let's since you mean let us.

File: docs/parallelism.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Paralellism
+
+`--parallelism` is a flag available on `rolling-update`.  This flag is better thought of as "chunking" a deployment, instead
```
Comment (davidxia) [inline docs/parallelism.md]: don't need the comma at the end.

Comment (davidxia) [reply docs/parallelism.md]: I prefer the word partition instead of chunk :) https://en.wikipedia.org/wiki/Partition_of_a_set

**Comment by bernhaker** [comment]
Good nob

**Comment by caipre** [comment]
LGTM.

File: docs/parallelism.md (hunk 2 above)
Comment (mavenraven) [reply docs/parallelism.md]: Great idea, thanks!

File: docs/parallelism.md (hunk 1 above)
Comment (mavenraven) [reply docs/parallelism.md]: thanks, fixed

File: docs/parallelism.md (hunk 3 above)
Comment (mavenraven) [reply docs/parallelism.md]: fixed

File: docs/parallelism.md (hunk 4 above)
Comment (mavenraven) [reply docs/parallelism.md]: 👍

**Pull request closed without merging**